File: 222-count-complete-tree-nodes/count-complete-tree-nodes.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int c=0;
    public int countNodes(TreeNode root) {
        preOrder(root);
       return c;
    }
    public void preOrder(TreeNode root){
        if(root==null) return;
        c++;
        preOrder(root.left);
        preOrder(root.right);
    }
}

File: 222-count-complete-tree-nodes/count-complete-tree-nodes.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int fun(TreeNode* root, int &count){
        if(root==nullptr) return 0;
        count++;
        int left=fun(root->left,count);
        int right=fun(root->right,count);
        return count;

    }
public:
    int countNodes(TreeNode* root) {
        if(root==nullptr) return 0;

        int count=0;
       return fun(root,count);
        //return count;
    }
};